### PR TITLE
created Composition and XRD for magento2 deploy in gcp with external …

### PR DESCRIPTION
**Pull request by ca7alindev**
…database, implemented gh action for build and push crossplane configuration packages under ghcr.io/web-seven/configuration-magento2-gcp repository

### CODE REVIEW

File: package/configuration-magento2-gcp/magento-gcp.yaml [hunk 1]
```diff
@@ -0,0 +1,706 @@
...
+          forProvider:
+            databaseVersion: MYSQL_5_7
+            deletionProtection: false
+            region: ""
+            settings:
+             - ipConfiguration:
+               - privateNetworkRef:
+                  name: ""
+            settings:
+              - diskSize: 20
```
Comment (evghen1) [inline package/configuration-magento2-gcp/magento-gcp.yaml:637]: @ca7alindev Add GCP persistent disc creation by current composition and use it here

File: package/configuration-magento2-gcp/magento-gcp.yaml [hunk 2]
```diff
@@ -0,0 +1,706 @@
...
+      base:
+        apiVersion: sql.gcp.upbound.io/v1beta1
+        kind: User
+        spec:
+          forProvider:
+            instanceSelector:
+              matchLabels: 
+                crossplane.io/external-name: ""
+            passwordSecretRef:
+              key: mariadb-password
```
Comment (evghen1) [inline package/configuration-magento2-gcp/magento-gcp.yaml:667]: @ca7alindev just simple "password'

File: package/configuration-magento2-gcp/magento-gcp.yaml [hunk 3]
```diff
@@ -0,0 +1,706 @@
...
+        toFieldPath: spec.forProvider.values.ingress.hostname
+
+
+    - name: db-instance-mysql
+      base:
+        apiVersion: sql.gcp.upbound.io/v1beta1
+        kind: DatabaseInstance
+        spec:
+          forProvider:
+            databaseVersion: MYSQL_5_7
```
Comment (evghen1) [inline package/configuration-magento2-gcp/magento-gcp.yaml:629]: @ca7alindev Make DB version configurable from XR

File: package/configuration-magento2-gcp/magento-gcp.yaml [hunk 4]
```diff
@@ -0,0 +1,706 @@
...
+        fromFieldPath: "status.sqlInstance.name"
+        toFieldPath: "spec.forProvider.instanceSelector.matchLabels[crossplane.io/external-name]"
+      - type: ToCompositeFieldPath
+        fromFieldPath: "metadata.annotations[crossplane.io/external-name]"
+        toFieldPath: "status.sqlInstance.database.name"
+
+
+  compositeTypeRef:
+    apiVersion: magentogke.webseven.io/v1alpha1
+    kind: MagentoGKE
```
Comment (evghen1) [inline package/configuration-magento2-gcp/magento-gcp.yaml:706]: @ca7alindev Move this to beginning of composition yaml

File: package/configuration-magento2-gcp/xrd.yaml [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+                gcp:
+                  type: object
+                  properties:
+                    secret:
+                      type: object
+                      properties:
+                        name:
+                          type: string
+                        namespace:
+                          type: string
```
Comment (evghen1) [inline package/configuration-magento2-gcp/xrd.yaml:42]: @ca7alindev Get rid from separate gcp section, just use providerConfigRef.name

File: package/configuration-magento2-gcp/xrd.yaml [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+                database:
+                  type: object
+                  properties:
+                    password:
+                      type: string
```
Comment (evghen1) [inline package/configuration-magento2-gcp/xrd.yaml:47]: @ca7alindev Please make password to be generated, not entered.

**Review by evghen1 — CHANGES_REQUESTED** [review]

File: package/configuration-magento2-gcp/magento-gcp.yaml (hunk 1 above)
Comment (ca7alindev) [reply package/configuration-magento2-gcp/magento-gcp.yaml:637]: @evghen1, persistent disk for databaseInstance?

File: .github/workflows/release-packages.yaml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+    - name: Set VERSION env
+      run: echo VERSION=$(echo ${GITHUB_REF} | rev | cut -d'/' -f 1 | rev ) >> ${GITHUB_ENV}
+
+    - name: Install crossplane CLI
+      id: crossplane
+      run: curl -sL "https://raw.githubusercontent.com/crossplane/crossplane/master/install.sh" | sh
+
+    - name: Build and push crossplane packages
+      run: ./crossplane xpkg build -f package/configuration-magento2-gcp -o configuration-magento2-gcp  && ./crossplane xpkg push -f configuration-magento2-gcp ${REGISTRY}/${{ github.repository }}/configuration-magento2-gcp:${{ env.VERSION }} || true' || true;
```
Comment (evghen1) [inline .github/workflows/release-packages.yaml:48]: @ca7alindev Please move all package file to "package" directory, what's idea of usage "configuration-magento2-gcp" subdirectory? This repository has just single configuration package.

File: package/configuration-magento2-gcp/xrd.yaml [hunk 8]
```diff
@@ -0,0 +1,194 @@
+apiVersion: apiextensions.crossplane.io/v1
+kind: CompositeResourceDefinition
+metadata:
+  name: magentogcps.gcp.web7.io
+spec:
+  group: gcp.web7.io
```
Comment (evghen1) [inline package/configuration-magento2-gcp/xrd.yaml:6]: @ca7alindev  gcp.web7.md

**Review by evghen1 — CHANGES_REQUESTED** [review]

**Comment by evghen1** [comment]
@ca7alindev Ready for review?

**Comment by ca7alindev** [comment]
> @ca7alindev Ready for review?
Yes

**Review by evghen1 — APPROVED** [review]

**Pull request merged**